src/wallet/wallet.h

```diff
@@ -841,6 +841,8 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
     CPubKey GenerateNewKey(WalletBatch& batch, bool internal = false) EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
     //! Adds a key to the store, and saves it to disk.
     bool AddKeyPubKey(const CKey& key, const CPubKey &pubkey) override EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
```

Comment:
Could make this method private and move next to AddWatchOnly:

https://github.com/bitcoin/bitcoin/blob/fb52d0684e0f03bca5d7ddcf3fc5b0657859ac2c/src/wallet/wallet.h#L692-L701